### PR TITLE
279 email verification switch to ses

### PR DESCRIPTION
**Pull request by omerzam**
# Description

Modified /verify/sendemail endpoint - switched from sending email via Mautic to SES. all other Mautic functionality stays the same.
Dependencies:
- Aws credentials

About # https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/goodserver/279

# How Has This Been Tested?

- Wrote a few unitests for the aws-ses component
- Made a postman request to the /verify/sendemail route (had to comment some of the existing code out to make the request work)

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/aws-ses/aws-ses.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// @flow
+import SES from 'aws-sdk/clients/ses'
+import conf from '../server.config'
+
+const accessKeyId = conf.awsSesAccessKey || null
```
Comment (sirpy) [inline src/server/aws-ses/aws-ses.js:5]: no need for '|| null'

File: src/server/verification/verificationAPI.js [hunk 2]
```diff
@@ -472,13 +473,15 @@ const setup = (app: Router, verifier: VerificationAPI, gunPublic: StorageAPI, st
...
-                mauticId: tempMauticId || mauticId
-              },
-              code
-            )
+            const { fullName } = userRec
+            if (!code || !fullName || !email) {
+              throw new Error('missing input for sending verification email')
+            }
+            const templateData = {
+              firstname: fullName,
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js:481]: use just first name, see in Mautic.sendVerificationEmail

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/server/verification/verificationAPI.js (hunk 2 above)
Comment (omerzam) [reply src/server/verification/verificationAPI.js:481]: i actually checked, Mautic.sendVerificationEmail uses full name and sends it as first name

File: src/server/aws-ses/aws-ses.js (hunk 1 above)
Comment (omerzam) [reply src/server/aws-ses/aws-ses.js:5]: I think it is best to validate the input without regard if the input was already validated somewhere else in the code
keeping the component as independent and decoupled as possible

**Pull request merged**